src/libraries/Microsoft.PowerFx.Core/Binding/INameResolver.cs

```diff
@@ -66,6 +66,15 @@ internal interface INameResolver
         // Looks up the parent control for the current context.
         bool LookupParent(out NameLookupInfo lookupInfo);
 
+        /// <summary>
+        /// In Canvas specifically, this is used to retrieve the full, derived, type
+        /// of a control object where the output schema may be dependent on a control inputs and heirarchy. 
+        /// All Non-Canvas hosts should not implement this method.
```

Comment:
>All Non-Canvas hosts should not implement this method. [](http://example.com/codeflow?start=12&length=54)

Since this is internal, will there by any non-Canvas/PowerApps host that can implement it (only PA has internals visibility)?

Reply:
The dataverse target also has internals visibility.